SwanProjects/swanprojects/handlers.py

```diff
@@ -0,0 +1,289 @@
+
+        old_name = ""
+        if "old_name" in input_data.keys():
+            old_name = input_data["old_name"]
+
+        old_stack = ""
+        if "old_stack" in input_data.keys():
+            old_stack = input_data["old_stack"]
+
+        old_platform = ""
+        if "old_platform" in input_data.keys():
+            old_platform = input_data["old_platform"]
+
+        old_release = ""
+        if "old_release" in input_data.keys():
+            old_release = input_data["old_release"]
+
+        old_userscript = ""
+        if "old_userscript" in input_data.keys():
+            old_userscript = input_data["old_userscript"]
+
+        name = input_data["name"]
+        stack = input_data["stack"]
+        platform = input_data["platform"]
+        release = input_data["release"]
+        user_script = input_data["user_script"]
+
+        project_dir = os.environ["HOME"] + "/SWAN_projects/" + name
+        if old_name != name:
+            try:
+                old_project_dir = os.environ["HOME"] + \
+                    "/SWAN_projects/" + old_name
+                os.rename(old_project_dir, project_dir)
+            except Exception as msg:
+                data = {"status": False, "project_dir": f"SWAN_projects/{old_name}",
+                        "msg": f"Error editing project folder {old_name},  traceback: {msg}"}
+                # this will stop the execution here, it's the same for the next exceptions.
+                self.finish(json.dumps(data))
+                return
+
+        if old_userscript != user_script:
+            userscript_file = project_dir + os.path.sep + '.userscript'
+            try:
+                with open(userscript_file, 'w') as f:
+                    f.write(user_script)
+                    f.close()
+            except Exception as msg:
+                data = {"status": False, "project_dir": f"SWAN_projects/{name}",
+                        "msg": f"Error editing .userscript for project {name},  traceback: {msg}"}
+                self.finish(json.dumps(data))
+                return
+
+        if stack != old_stack or platform != old_platform or release != old_release or corrupted:
+            swan_project_file = project_dir + os.path.sep + '.swanproject'
+            swan_project_content = {'stack': stack, 'release': release,
+                                    'platform': platform}
+            kernel_dir = project_dir + "/.local/share/jupyter/kernels"
+
+            # removing old native kernels for python only(this is generated by us)
+            if os.path.exists(kernel_dir + "/python2"):
+                shutil.rmtree(kernel_dir + "/python2")
+
+            if os.path.exists(kernel_dir + "/python3"):
+                shutil.rmtree(kernel_dir + "/python3")
+
+            with open(swan_project_file, 'w+') as f:
+                f.write(json.dumps(swan_project_content,
+                        indent=4, sort_keys=True))
+                f.close()
+            command = get_env_isolated()
```

Comment:
I've seen you using popen in many places and doing always the same. So it looks like you are repeating the code.
Just create a method somewhere and use it everywhere.

Reply:
done